components/raftstore-v2/src/operation/query/local.rs

```diff
@@ -142,39 +345,339 @@ where
     }
 }
 
+struct SnapRequestInspector<'r> {
+    delegate: &'r ReadDelegate,
+    logger: &'r Logger,
+}
+
+impl<'r> RequestInspector for SnapRequestInspector<'r> {
+    fn inspect(&mut self, req: &RaftCmdRequest) -> Result<RequestPolicy> {
```

Comment:
What's the point of implementing `RequestInspector`?

Reply:
Remove unnecessary checks, and assert for the snap request.

Reply:
What's the difference between implementing as methods instead of trait?